src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/GetAzureContainerRegistry.cs

```diff
@@ -14,65 +14,87 @@
+            if (string.Equals(ParameterSetName, ListRegistriesParameterSet))
             {
-                var registry = RegistryClient.GetRegistry(ResourceGroupName, Name);
-                WriteObject(new PSContainerRegistry(registry));
+                ListRegistry();
             }
             else
             {
-                List<PSContainerRegistry> list = new List<PSContainerRegistry>();
-
-                var registries = RegistryClient.ListRegistries(ResourceGroupName);
-                foreach (Registry registry in registries)
+                if (MyInvocation.BoundParameters.ContainsKey("ResourceId") || !string.IsNullOrWhiteSpace(ResourceId))
                 {
-                    list.Add(new PSContainerRegistry(registry));
+                    string resourceGroup, registryName, childResourceName;
+                    if(!ConversionUtilities.TryParseRegistryRelatedResourceId(ResourceId, out resourceGroup, out registryName, out childResourceName))
+                    {
+                        WriteInvalidResourceIdError(InvalidRegistryResourceIdErrorMessage);
+                        return;
+                    }
+
+                    ResourceGroupName = resourceGroup;
+                    Name = registryName;
                 }
+                ShowRegistry();
+            }
+        }
+
+        private void ShowRegistry()
+        {
+            var registry = RegistryClient.GetRegistry(ResourceGroupName, Name);
+            var psRegistry = new PSContainerRegistry(registry);
+
+            if (IncludeDetail)
+            {
+                SetRegistryDetials(psRegistry);
+            }
+            WriteObject(psRegistry);
+        }
 
-                while (!string.IsNullOrEmpty(registries.NextPageLink))
+        private void ListRegistry()
+        {
+            var psRegistries = RegistryClient.ListAllRegistries(ResourceGroupName);
+
+            if (IncludeDetail)
+            {
+                foreach(var psr in psRegistries)
                 {
-                    registries = RegistryClient.ListRegistriesUsingNextLink(ResourceGroupName, registries.NextPageLink);
-                    foreach (Registry registry in registries)
-                    {
-                        list.Add(new PSContainerRegistry(registry));
-                    }
+                    SetRegistryDetials(psr);
                 }
+            }
+            WriteObject(psRegistries, true);
+        }
 
-                WriteObject(list, true);
+        private void SetRegistryDetials(PSContainerRegistry registry)
+        {
+            if(!string.IsNullOrEmpty(registry.ResourceGroupName) && !string.IsNullOrEmpty(registry.Name))
+            {
+                registry.Usages = RegistryClient.ListRegistryUsage(registry.ResourceGroupName, registry.Name)?.Value;
+                registry.Replications = RegistryClient.ListAllReplications(registry.ResourceGroupName, registry.Name);
```

Comment:
I would consider including `credential` here to be consistent with `Storage` and get rid of `replications`.